syntax = "proto3";

package masterworker;

service MasterWorker {
	rpc assignMapReduce (WorkerRequest) returns (WorkerReply) {}
}

message WorkerRequest {
	bool do_map = 1;
	string map = 2;
	repeated string reduce = 3;
	int32 n_output = 4;
	string output_dir = 5;
	string user_id = 6;
}

message WorkerReply {
	repeated string map = 1;
	string reduce = 2;
}
